Mathlib/CategoryTheory/Sites/Equivalence.lean

```diff
@@ -0,0 +1,231 @@
+
+namespace Equivalence
+
+variable {C : Type*} [Category C] (J : GrothendieckTopology C)
+variable {D : Type*} [Category D] (e : C ≌ D)
+variable (A : Type*) [Category A]
+
+theorem locallyCoverDense : LocallyCoverDense J e.inverse := by
+  intro X T
+  convert T.prop
+  ext Z f
+  constructor
+  · rintro ⟨_, _, g', hg, rfl⟩
+    exact T.val.downward_closed hg g'
+  · intro hf
+    refine ⟨e.functor.obj Z, (Adjunction.homEquiv e.toAdjunction _ _).symm f, e.unit.app Z, ?_, ?_⟩
+    · simp only [Adjunction.homEquiv_counit, Functor.id_obj, Equivalence.toAdjunction_counit,
+        Sieve.functorPullback_apply, Presieve.functorPullback_mem, Functor.map_comp,
+        Equivalence.inv_fun_map, Functor.comp_obj, Category.assoc, Equivalence.unit_inverse_comp,
+        Category.comp_id]
+      exact T.val.downward_closed hf _
+    · simp
+
+theorem coverPreserving :
+    CoverPreserving J (e.locallyCoverDense J).inducedTopology e.functor where
+  cover_preserve {U S} h := by
+    change _ ∈ J.sieves (e.inverse.obj (e.functor.obj U))
+    convert J.pullback_stable (e.unitInv.app U) h
+    ext Z f
+    rw [← Sieve.functorPushforward_comp]
+    simp only [Sieve.functorPushforward_apply, Presieve.functorPushforward, exists_and_left, id_obj,
+      comp_obj, Sieve.pullback_apply]
+    constructor
+    · rintro ⟨W, g, hg, x, rfl⟩
+      rw [Category.assoc]
+      apply S.downward_closed
+      simpa using S.downward_closed hg _
+    · intro hf
+      exact ⟨_, e.unitInv.app Z ≫ f ≫ e.unitInv.app U, S.downward_closed hf _,
+        e.unit.app Z ≫ e.unit.app _, (by simp)⟩
+
+instance : IsCoverDense e.functor (e.locallyCoverDense J).inducedTopology where
+  is_cover U := by
+    change _ ∈ J.sieves _
+    convert J.top_mem (e.inverse.obj U)
+    ext Y f
+    simp only [Sieve.functorPushforward_apply, Presieve.functorPushforward, exists_and_left,
+      Sieve.top_apply, iff_true]
+    exact ⟨e.functor.obj Y, (Adjunction.homEquiv e.toAdjunction _ _).symm f,
+      Presieve.in_coverByImage _ _, e.unit.app _, (by simp)⟩
+
+instance : IsContinuous e.functor J (e.locallyCoverDense J).inducedTopology :=
+  IsCoverDense.isContinuous _ _ _ (e.coverPreserving J)
+
+instance : IsCoverDense e.inverse J where
+  is_cover U := by
+    convert J.top_mem U
+    ext Y f
+    simp only [Sieve.functorPushforward_apply, Presieve.functorPushforward, exists_and_left,
+      Sieve.top_apply, iff_true]
+    let g : e.inverse.obj _ ⟶ U := (e.unitInv.app Y) ≫ f
+    have : (Sieve.coverByImage e.inverse U).arrows g := Presieve.in_coverByImage _ g
+    replace := Sieve.downward_closed _ this (e.unit.app Y)
+    simpa using this
+
+instance : IsContinuous e.inverse (e.locallyCoverDense J).inducedTopology J :=
+  IsCoverDense.isContinuous _ _ _ (e.locallyCoverDense J).inducedTopology_coverPreserving
+
+/-- The functor in the equivalence of sheaf categories. -/
+@[simps!]
+def sheafCongr_functor : Sheaf J A ⥤ Sheaf (e.locallyCoverDense J).inducedTopology A where
```

Comment:
In this situation, we should use a name like `sheafCongrFunctor` or, better, `sheafCongr.functor`, for the auxiliary definitions.